presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PlanNodeSearcher.java

```diff
@@ -125,7 +125,7 @@ public PlanNodeSearcher skipOnlyWhen(Predicate<PlanNode> skipOnly)
         if (where.test(node)) {
             nodes.add((T) node);
         }
-        if (skipOnly.test(node)) {
+        if (skipOnly.test(node) || where.test(node)) {
```

Comment:
Yes, but only if you call `skipOnlyWhen(..)`. Otherwise `skipOnly` defaults to true.